src/deavinaSJun20/ValidatePhoneNumber.java

```diff
@@ -0,0 +1,85 @@
+package deavinaSJun20;
+
+public class ValidatePhoneNumber {
+
+	boolean flag = false;
+
+	public void validPhoneNo(String phoneNo) {
+		switch (phoneNo.length()) {
+		case 14:
+			checkFourteenDigit(phoneNo);
+			break;
+		case 11:
+			checkElevenDigit(phoneNo);
+			break;
+		case 12:
+			checkTwelveDigit(phoneNo);
+			break;
+		case 10:
+			checkTenDigit(phoneNo);
+			break;
+		default:
+			System.out.println(phoneNo + " is in valid phone no");
+		}
+	}
+
+	public void displayResult(String phoneNo, boolean flag) {
+		if (flag == true)
+			System.out.println(phoneNo + " is a valid phone no");
+		else
+			System.out.println(phoneNo + " is an invalid phone no");
+	}
+
+	public void checkFourteenDigit(String phoneNo) {
+		for (int index = 5; index <= 13; index++) {
+			if (phoneNo.charAt(0) == '+' && phoneNo.charAt(1) == '9' && phoneNo.charAt(2) == '1'
+					&& phoneNo.charAt(3) == ' '
+					&& (phoneNo.charAt(4) == '9' || phoneNo.charAt(4) == '8' || phoneNo.charAt(4) == '7')
+					&& (Character.isDigit(phoneNo.charAt(index)))) {
+				displayResult(phoneNo, true);
+				break;
+			}
+		}
+	}
+
+	public void checkElevenDigit(String phoneNo) {
+		for (int index = 2; index <= 11; index++) {
+			if (phoneNo.charAt(0) == '0'
+					&& (phoneNo.charAt(1) == '9' || phoneNo.charAt(1) == '8' || phoneNo.charAt(1) == '7')
+					&& (Character.isDigit(phoneNo.charAt(index)))) {
+				displayResult(phoneNo, true);
+				break;
+			}
+		}
+	}
+
+	private void checkTwelveDigit(String phoneNo) {
+		if ((phoneNo.charAt(0) == '9' || phoneNo.charAt(0) == '8' || phoneNo.charAt(0) == '7')
+				&& (Character.isDigit(phoneNo.charAt(1))) && (Character.isDigit(phoneNo.charAt(2)))
```

Comment:
code is working fine, please try to use substring instead of logical operator.

Reply:
No logical impact using substring, this is also fine